### PR TITLE
chore: migrate entrypoint of CLI to esm modules

### PR DESCRIPTION
**Pull request by lukasholzer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

As the esm module migration is such a big thing I try to split it up into small chunks and try to use both cjs and esm in a grace period to do a incremental migration.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

**Comment by ehmicky** [comment]
CI tests are currently failing. I am wondering if maybe the new tests setup will help with this?

File: tests/eleventy-site/_site/_redirects [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/force              /test                           200!
```
Comment (erezrokah) [inline tests/eleventy-site/_site/_redirects:1]: I believe these files are redundant? They were moved under `tests/integration` in https://github.com/netlify/cli/pull/4164

**Comment by erezrokah** [comment]
Hi @lukasholzer, I left a small comment.

Also in order to trigger a release we'll need to merge this as `fix` or `feat`.

**Comment by janosh** [comment]
Always great to see more things migrating to ESM! Has this PR fallen under the radar?

**Comment by lukasholzer** [comment]
@janosh sadly I had to put it on the side due to a shift of resources and priorities but If you want to continue feel free to! 

I would love to see this effort going further!

**Comment by janosh** [comment]
I'm not familiar with the code base so I could only take over if it's minor stuff. What's left to do beside merge conflict resolution and addressing the comment by @erezrokah?

**Comment by lukasholzer** [comment]
@janosh that should be all - having passing checks (e2e tests and unit tests) should provide the confidence

**Comment by lukasholzer** [comment]
Basically we try to split it from top to bottom to migrate small parts for small parts as you can import commonjs in esm.

But vice verca is way more complicate through dynamic imports we can go from top to bottom and move the files to .mjs that import some commonjs.

The downside is that we have to move named imports then to default imports and destruct it later but that is all.

**Comment by danez** [comment]
Closing as a lot of this is now already in the main branch

**Pull request closed without merging**